uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -154,6 +161,25 @@ impl APIConverter<Enum> for weedle::InterfaceDefinition<'_> {
                 })
                 .collect::<Result<Vec<_>>>()?,
             flat: false,
+            ..Default::default()
+        })
+    }
+}
+
+impl APIConverter<Enum> for &syn::ItemEnum {
+    fn convert(&self, ci: &mut ComponentInterface) -> Result<Enum> {
+        let attrs = super::synner::Attributes::try_from(&self.attrs)?;
+        let variants = self
+                .variants
+                .iter()
+                .map(|v| v.convert(ci))
+                .collect::<Result<Vec<_>>>()?;
+        let flat = ! variants.iter().any(|v| v.fields().len() > 0);
+        Ok(Enum {
+            name: self.ident.to_string(),
+            variants,
+            flat,
+            docs: attrs.docs,
```

Comment:
Here, I've taken the same `APIConverter` pattern that we've been using for `weedle` parse nodes and have implemented it for `syn` parse nodes as well. It more-or-less Just Worked. Like what we currently do for weedle, the trickiest part is looking for all the extra bits of syntax that we don't want to support and erroring out if we see any.

For example, this takes a parsed Rust `enum` definition and converts it into a definition for a UniFFI component `enum`.